Can't build from source on the OpenSolaris/illumos

## Description of the problem or steps to reproduce

Cannot build from source due to following error:

```
ivan@hipster:~/pro/tp/micro$ gmake
GOOS=illumos GOARCH=amd64 go generate ./runtime
CGO_ENABLED=0 go build -trimpath -ldflags "-s -w -X github.com/zyedidia/micro/v2/internal/util.Version=2.0.15-dev.90 -X github.com/zyedidia/micro/v2/internal/util.CommitHash=2898f159 -X 'github.com/zyedidia/micro/v2/internal/util.CompileDate=December 22, 2024' """ ./cmd/micro
# github.com/zyedidia/terminal
../../../go/pkg/mod/github.com/zyedidia/terminal@v0.0.0-20230315200948-4b3bcf6dddef/ioctl_posix.go:13:11: undefined: syscall.SYS_IOCTL
gmake: *** [Makefile:27: build-quick] Error 
```

Some additional information on the topic can be found here
https://stackoverflow.com/questions/73280640
Please update from "syscall" to "golang.org/x/sys/unix"

## Specifications

Commit hash: 2898f1590d7a0588c5ae27753b2d68bd2607156a
OS: SunOS hipster 5.11 illumos-a2057e4a8d i86pc i386 i86pc
Terminal:

--- a/go.mod
+++ b/go.mod
@@ -9,7 +9,7 @@ require (
 	github.com/mattn/go-runewidth v0.0.16
 	github.com/micro-editor/json5 v1.0.1-micro
 	github.com/micro-editor/tcell/v2 v2.0.11
-	github.com/micro-editor/terminal v0.0.0-20250105114944-ffd0fc59e777
+	github.com/micro-editor/terminal v0.0.0-20250324214352-e587e959c6b5
 	github.com/mitchellh/go-homedir v1.1.0
 	github.com/sergi/go-diff v1.1.0
 	github.com/stretchr/testify v1.4.0
@@ -29,7 +29,7 @@ require (
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	github.com/rivo/uniseg v0.2.0 // indirect
 	github.com/zyedidia/poller v1.0.1 // indirect
-	golang.org/x/sys v0.6.0 // indirect
+	golang.org/x/sys v0.28.0 // indirect
 )
 
 replace github.com/kballard/go-shellquote => github.com/micro-editor/go-shellquote v0.0.0-20250101105543-feb6c39314f5

--- a/internal/action/terminal_supported.go
+++ b/internal/action/terminal_supported.go
@@ -1,4 +1,4 @@
-//go:build linux || darwin || dragonfly || openbsd_amd64 || freebsd
+//go:build linux || darwin || dragonfly || solaris || openbsd || netbsd || freebsd
 
 package action
 

--- a/internal/action/terminal_unsupported.go
+++ b/internal/action/terminal_unsupported.go
@@ -1,4 +1,4 @@
-//go:build !linux && !darwin && !freebsd && !dragonfly && !openbsd_amd64
+//go:build plan9 || nacl || windows
 
 package action
 

--- a/tools/cross-compile.sh
+++ b/tools/cross-compile.sh
@@ -74,6 +74,16 @@ echo "Linux ARM 64"
 GOOS=linux GOARCH=arm64 make build
 create_artefact_generic "linux-arm64"
 
+# Solaris
+echo "Solaris 64"
+GOOS=solaris GOARCH=amd64 make build
+create_artefact_generic "solaris64"
+
+# Illumos
+echo "Illumos 64"
+GOOS=illumos GOARCH=amd64 make build
+create_artefact_generic "illumos64"
+
 # NetBSD
 echo "NetBSD 64"
 GOOS=netbsd GOARCH=amd64 make build

